plugins/services/src/js/reducers/serviceForm/JSONReducers/EnvironmentVariables.js

```diff
@@ -63,20 +63,26 @@ module.exports = {
       return [];
     }
 
-    return Object.keys(state.env).reduce(function(memo, key, index) {
-      /**
+    return Object.keys(state.env)
+      .filter(function(key) {
+        return typeof state.env[key] === "string";
```

Comment:
We could instead of the filter do this in the reduce like 

```JS
if(typeof state.env[key] === "string") {
  return memo;
}
```
With this we would remove half of the cycles above the env vars. With the same result.